pkg/gen/nodejs-templates/helm.ts.mustache

```diff
@@ -63,63 +63,76 @@ export namespace v2 {
-
-            try {
-                let chart: string;
-                let defaultValues: string;
-                if (isChartOpts(config)) {
-                    // Fetch chart.
-                    fetch(`${config.repo}/${config.chart}`, {
-                        destination: chartDir.name,
-                        version: config.version
-                    });
-                    chart = path.quotePath(nodepath.join(chartDir.name, config.chart));
-                    defaultValues = path.quotePath(
-                        nodepath.join(chartDir.name, config.chart, "values.yaml")
-                    );
-                } else {
-                    chart = path.quotePath(config.path);
-                    defaultValues = path.quotePath(nodepath.join(chart, "values.yaml"));
-                }
-
-                // Write overrides file.
-                const data = JSON.stringify(config.values || {}, undefined, "  ");
-                fs.writeFileSync(overrides.name, data);
-
-                // Does not require Tiller. From the `helm template` documentation:
-                //
-                // >  Render chart templates locally and display the output.
-                // >
-                // > This does not require Tiller. However, any values that would normally be
-                // > looked up or retrieved in-cluster will be faked locally. Additionally, none
-                // > of the server-side testing of chart validity (e.g. whether an API is supported)
-                // > is done.
-                const release = shell.quote([releaseName]);
-                const values = path.quotePath(overrides.name);
-                const namespaceArg = config.namespace
-                    ? `--namespace ${shell.quote([config.namespace])}`
-                    : "";
-                const yamlStream = execSync(
-                    `helm template ${chart} --name ${release} --values ${defaultValues} --values ${values} ${namespaceArg}`
-                ).toString();
-                this.resources = this.parseTemplate(yamlStream, config.transformations);
-            } catch (e) {
-                // Shed stack trace, only emit the error.
-                throw new pulumi.RunError(e.toString());
-            } finally {
-                // Clean up temporary files and directories.
-                chartDir.removeCallback();
-                overrides.removeCallback();
-            }
+			const allConfig = pulumi.output(config);
+			const configDeps = Array.from(<Set<pulumi.Resource>>(<any>allConfig).resources());
+
+			(<any>allConfig).isKnown.then((isKnown: boolean) => {
+				if (!isKnown) {
+					// Note that this can only happen during a preview.
+					pulumi.log.info("[Can't preview] all chart values must be known ahead of time to generate an accurate preview.", this);
+				}
+			});
+
+			this.resources = allConfig.apply(cfg => {
+				// Create temporary directories and files to hold chart data and override values.
```

Comment:
oh, eek, I almost missed this -- these are tabs. Can we please double-check the formatting to make sure it's consistent with what we have? :)

Reply:
I'll fix this now--got a little itchy on the trigger.